### ISSUE TITLE
Fix how we verify the project status

### ISSUE THREAD

**Issue by lorenzo-cavazzi** [SwissDataScienceCenter/renku-ui#1858]
## Description

When visiting some of the project tabs (e.g. status overview or datasets) we invoke the `/renku/project.lock_status` API to verify whether the user is already modifying the project. If that is the case, we show a label next to the title.
This currently has 2 problems.

* We occasionally invoke the API multiple times in a row (E.G. when using a URL leading directly to the `/projects/<id>/datasets` page). This triggers the project clone on the backend and locks it for some time, depending on the project size. The last API invocation returns a locked status even if that is a transitory state.
![Screenshot_20220520_090140](https://user-images.githubusercontent.com/43481553/169488760-2e8d796a-1f4f-4639-b57b-1535e2dfd885.png)

* The anonymous users can't write to the project, and the API returns an error. We show a confusing message. ![image (5)](https://user-images.githubusercontent.com/43481553/169488826-c690c544-a9d9-4f56-b388-81e9dccde7ac.png)

## Fix

We should:

* [ ] invoke the API only once
* [ ] do not invoke it (nor show any warning) for anonymous users
* [ ] _extra (maybe follow-up issue?): use `setTimeout` to check the status again after a few seconds when the project is locked_
